Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3720,6 +3720,27 @@ LayoutUnit RenderBox::constrainBlockMarginInAvailableSpaceOrTrim(const RenderBox
     return marginSide == MarginTrimType::BlockStart ? minimumValueForLength(style().marginBeforeUsing(&containingBlock.style()), availableSpace) : minimumValueForLength(style().marginAfterUsing(&containingBlock.style()), availableSpace);
 }
 
+bool RenderBox::hasTrimmedMargin(MarginTrimType marginTrimType) const
+{
+    ASSERT(!needsLayout());
+    auto* container = containingBlock();
+
+    // FIXME: Remove these asserts as they are implemented for each layout context
+    if (isFlexItem() || isGridItem())
+        ASSERT_NOT_IMPLEMENTED_YET();
+    else if (container->isBlockContainer())
+        ASSERT(marginTrimType == MarginTrimType::BlockStart);
+    else
+        return false;
```

Comment:
please assert when this function is called with something other than block-start.

Reply:
I think I made the change you were referring to (the last false return), but let me know if not